### ISSUE TITLE
Consumer groups not compatible with kafka-consumer-groups.sh?

### ISSUE THREAD

**Issue by shanson7** [edenhill/librdkafka#660]
Description
===========
Consumer groups in rdkafka seem to cause issues with the kafka consumer groups tools. This may very well be an issue in kafka-consumer-groups.sh script, but perhaps you can help shed some light.

How to reproduce
================

Spin up a consumer and let it store offsets in the brokers.
```
$ bin/kafka-consumer-groups.sh --new-consumer --bootstrap-server localhost:9092 --describe --group sean_test
Error while executing consumer group command Error reading field 'user_data': java.lang.IllegalArgumentException
org.apache.kafka.common.protocol.types.SchemaException: Error reading field 'user_data': java.lang.IllegalArgumentException
        at org.apache.kafka.common.protocol.types.Schema.read(Schema.java:71)
        at org.apache.kafka.clients.consumer.internals.ConsumerProtocol.deserializeAssignment(ConsumerProtocol.java:109)
        at kafka.admin.AdminClient$$anonfun$describeConsumerGroup$1.apply(AdminClient.scala:165)
        at kafka.admin.AdminClient$$anonfun$describeConsumerGroup$1.apply(AdminClient.scala:164)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
        at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:244)
        at scala.collection.immutable.List.foreach(List.scala:318)
        at scala.collection.TraversableLike$class.map(TraversableLike.scala:244)
        at scala.collection.AbstractTraversable.map(Traversable.scala:105)
        at kafka.admin.AdminClient.describeConsumerGroup(AdminClient.scala:164)
        at kafka.admin.ConsumerGroupCommand$KafkaConsumerGroupService.describeGroup(ConsumerGroupCommand.scala:314)
        at kafka.admin.ConsumerGroupCommand$ConsumerGroupService$class.describe(ConsumerGroupCommand.scala:84)
        at kafka.admin.ConsumerGroupCommand$KafkaConsumerGroupService.describe(ConsumerGroupCommand.scala:302)
        at kafka.admin.ConsumerGroupCommand$.main(ConsumerGroupCommand.scala:63)
        at kafka.admin.ConsumerGroupCommand.main(ConsumerGroupCommand.scala)
```

Checklist
=========
Please provide the following information:

 - [x] librdkafka version (release number or git tag): master
 - [x] Apache Kafka version: 0.9.0
 - [ ] librdkafka client configuration:
 - [x] Operating system: rhel7.2
 - [ ] Using the legacy Consumer
 - [x] Using the high-level KafkaConsumer
 - [ ] Provide logs (with `debug=..` as necessary) from librdkafka
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by edenhill** [comment]
This is a bug in broker 0.9.0.x, it is fixed in 0.10.0.0 (soon to be released).
But I created a workaround in librdkafka, what librdkafka version are you using?

**Comment by shanson7** [comment]
This is master from...a while ago :)
I'll pull and try. Is there a tag/branch I should pin to?

**Comment by shanson7** [comment]
Ah, actually, this is master from 7 days ago.

**Comment by edenhill** [comment]
I was wrong, the workaround was for a similar issue. The consumer-protocol userdata is still sent as NULL and this is not supported by broker 0.9.0.0. It seems to be fixed in broker 0.9.0.1 though.

**Comment by shanson7** [comment]
Oh, excellent. Thank you again!
